### PR TITLE
Mark buffers as changed after cancelling :confirm qall

### LINKED ISSUE #14593: flag modified of a modified buffer not set after canceling quit operation
- `nvim`: v0.4.4, v0.5.0-dev+1339-g5d9c56012

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC --cmd 'set hidden'
```
open 2 files and modify them
don't save the buffers
```
:confirm qall
```
answer no to the first question and cancel to the second
go back to the previous buffer

### Actual behaviour

the flag modified is not set even though the buffer is modified

### Expected behaviour

the buffer must have the modified flag set

### PR DESCRIPTION
**Pull request by chrisbra**
When doing `:confirm :qall` with multiple modified buffers and typing
`N` for the first changed buffer and then `C`ancelling the dialog, the
first modified buffer already marked unchanged/unmodified.

This means:

1) There is a difference between the buffer and the file on disk
2) Vim will happily throw away those changes when quitting

So let's handle `N`o specifically and remember which buffers have been
marked unmodified (in order to quit properly). And in case the confirm
dialog is canceled, iterate over those buffers and mark them as modified
again.

fixes: neovim/neovim#14593

### CODE REVIEW

File: src/testdir/test_excmd.vim [hunk 1]
```diff
@@ -597,4 +599,45 @@ func Test_command_not_implemented_E319()
...
+  call term_wait(buf, 500)
+  call term_sendkeys(buf, ":ls\n")
+  call term_wait(buf, 500)
+  "sleep 500m
+  call VerifyScreenDump(buf, 'Test_excmd_confirm_1', {})
+
+  "call WaitForAssert({-> assert_match('\d\s*[%#][ah]\s*+\s*"X', term_getline(buf, 19))}, 1000)
+  "call WaitForAssert({-> assert_match('\d\s*%\?a\s*+\s*"Xfoo"', term_getline(buf, 18))}, 2000)
+  "call WaitForAssert({-> assert_match('\d\s*[%#][ah]\s*+\s*"X', term_getline(buf, 19))}, 1000)
+  "call WaitForAssert({-> assert_match('\d\s*#\?h\s*+\s*"Xbar"', term_getline(buf, 19))}, 2000)
```
Comment (dpelle) [inline src/testdir/test_excmd.vim]: Why are these lines commented?

Comment (chrisbra) [reply src/testdir/test_excmd.vim]: Me trying to use match the output of `:ls`. For some reasons it did not work, so I left them commented out and used a screen test instead. I think the issue was a missing `term_wait()` call. 

Will remove those comments.

**Comment by brammool** [comment]
Christian wrote:

> When doing `:confirm :qall` with multiple modified buffers and typing
> `N` for the first changed buffer and then `C`ancelling the dialog, the
> first modified buffer already marked unchanged/unmodified.
> 
> This means:
> 
> 1) There is a difference between the buffer and the file on disk
> 2) Vim will happily throw away those changes when quitting
> 
> So let&#39;s handle `N`o specifically and remember which buffers have been
> marked unmodified (in order to quit properly). And in case the confirm
> dialog is canceled, iterate over those buffers and mark them as modified
> again.
> 
> fixes: neovim/neovim#14593

If the user answers "N"/No for a buffer, one can assume that the changes
in this buffer are irrelevant.  If then using "C"/Cancel for another
buffer, this is probably because the user wants to have another look at
this one.  Does this mean that the previous "N" should be ignored?  I
don't think so.

The most usual sequence would be:

- User tries to quit Vim while there are modified buffers.
- Dialog shows for each modified buffer.  User selects "N" for buffers
  that are irrelevant.
- User sees a buffer that he/she isn't sure about, selects "C" to have a
  closer look.
- After inspection, closes Vim again.

Does the user want to go over all the buffers again?  I would think not,
those where the response was "N" can be ignored.  Having to go over all
modified buffers again is slightly annoying.

-- 
FATHER:    You only killed the bride's father - that's all -
LAUNCELOT: Oh dear, I didn't really mean to...
FATHER:    Didn't mean to?  You put your sword right through his head!
LAUNCELOT: Gosh - Is he all right?
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
Hm okay, different mindset. I would expect after a `cancel` everything be like before running `:q` command.

**Pull request closed without merging**